packages/registry/src/index.ts

```diff
@@ -0,0 +1,173 @@
+import { isArray } from 'lodash';
+import * as configTypes from './types';
+
+type ConfigData = {
+  settings: configTypes.SettingsConfig;
+  blocks: configTypes.BlocksConfig;
+  views: configTypes.ViewsConfig;
+  widgets: configTypes.WidgetsConfig;
+  addonReducers: configTypes.AddonReducersConfig;
+  addonRoutes: configTypes.AddonRoutesConfig;
+  slots: configTypes.SlotsConfig;
+  components: configTypes.ComponentsConfig;
+  experimental: configTypes.ExperimentalConfig;
+};
+
+class Config {
+  public _data: ConfigData | Record<string, never>;
+  static instance: InstanceType<typeof Config>;
+
+  constructor() {
+    if (!Config.instance) {
+      this._data = {};
+      Config.instance = this;
+    }
+
+    return Config.instance;
+  }
+
+  set(registry: keyof ConfigData, item: ConfigData[keyof ConfigData]) {
```

Comment:
I`m not sure, I`m only on github web-view, but try:

`set<RegistryKey extends keyof ConfigData>(registry: RegistryKey, item: ConfigData[RegistryKey]) {`

Reply:
It worked! What's the rationale behind?

Reply:
@sneridagh Nice! This ist a Generic. Something I miss in mypy in this extent. Sorry I'm not really good in explaining things in an easy way, but TypeScript docs are really good.

https://www.typescriptlang.org/docs/handbook/2/generics.html

Please don't hesitate to ping me, if you face an other TypeScript problem. As you did with the comment or also "as any" ist your friend. I do this very frequently and come back later.